### PR TITLE
Remove distortion on HMD previews, both mono and stereo

### PR DESCRIPTION
**Pull request by jherico**
* Migrate both the OpenVR and Oculus plugins to share a base class for common functionality like monoscopic preview
* Letterbox the on-screen preview display so that it is without distortion (both mono and stereo preview)

### CODE REVIEW

**Comment by jherico** [comment]
fix for https://app.asana.com/0/27650181942747/90528769008005 and https://app.asana.com/0/27650181942747/90523169233092

**Comment by hifi-gustavo** [comment]
Build console output for commit: 56fc9ea3016e8cc6589d1f0d00c9b4239bb7f92c available at: https://gist.github.com/2d2bb5ad5650207d4e4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.dmg

**Comment by judasshuffle** [comment]
Does remove the distortion from the preview, much nicer.
Will improve chris's  rift wearing meeting videos lol

File: libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+#include <glm/gtc/matrix_transform.hpp>
+
+#include <QtCore/QLoggingCategory>
+#include <QtWidgets/QWidget>
+
+#include <GLMHelpers.h>
+#include <plugins/PluginContainer.h>
+
+Q_DECLARE_LOGGING_CATEGORY(displayplugins)
+Q_LOGGING_CATEGORY(displayplugins, "hifi.displayplugins")
```
Comment (hyperlogic) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: Why not follow the existing pattern of putting `Q_DECLARE_LOGGING_CATEGORY()` in a displayPluginsLogging.h and put the `Q_LOGGING_CATEGORY()` in displayPluginsLogging.cpp?

File: libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+//    drawUnitQuad();
+//}
+///*
+//The swapbuffer call here is only required if we want to mirror the content to the screen.
+//However, it should only be done if we can reliably disable v-sync on the mirror surface,
+//otherwise the swapbuffer delay will interefere with the framerate of the headset
+//*/
+//if (_enablePreview) {
+//    swapBuffers();
+//}
```
Comment (hyperlogic) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: please remove dead code

**Comment by hyperlogic** [comment]
This is not working very well on OpenVR/Vive Pre.

* The scene culling seems to be affected.  For example, in demo the dice table is not visible until you walk right up to it and it pops into view.
* The 2d display shows the cropped quad, however it flashes.  Some frames the screen is all black, some frames the screen shows the quad as expected.

**Comment by hyperlogic** [comment]
I would attach a screen shot, but all the screenshots for vive turn up all black.  Missing a swap buffer?

File: libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp (hunk 1 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: That was just copy pasted from an one of the display plugins.  I'll fix.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6c99a593aa59b978a0a9af6c0d90972c8c873295 available at: https://gist.github.com/2d2bb5ad5650207d4e4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d61f4fbb253d995fa1e83c86e1fa3a1b17b8e38c available at: https://gist.github.com/2d2bb5ad5650207d4e4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 585e0ec7635387c860c7886a2e2561a34fcb8a93 available at: https://gist.github.com/2d2bb5ad5650207d4e4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d1583e8cbe601cb736c154e7dbfaae24a1fbdc1a available at: https://gist.github.com/2d2bb5ad5650207d4e4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.dmg

**Comment by hyperlogic** [comment]
So, the culling is now working properly for the Vive, however, the 2d display is still flashing.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8289c39a2ab15aa5fa849316f51d82b417dca1ea available at: https://gist.github.com/2d2bb5ad5650207d4e4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7118.dmg

**Comment by hyperlogic** [comment]
:+1:

**Pull request merged**